packages/adapter-cloudflare-workers/files/entry.js

```diff
@@ -86,19 +91,29 @@ export default {
  * @param {any} context
  */
 async function get_asset_from_kv(req, env, context, map = mapRequestToAsset) {
-	return await getAssetFromKV(
-		{
-			request: req,
-			waitUntil(promise) {
-				return context.waitUntil(promise);
+	try {
+		return await getAssetFromKV(
+			{
+				request: req,
+				waitUntil(promise) {
+					return context.waitUntil(promise);
+				}
+			},
+			{
+				ASSET_NAMESPACE: env.__STATIC_CONTENT,
+				ASSET_MANIFEST: static_asset_manifest,
+				mapRequestToAsset: map
 			}
-		},
-		{
-			ASSET_NAMESPACE: env.__STATIC_CONTENT,
-			ASSET_MANIFEST: static_asset_manifest,
-			mapRequestToAsset: map
+		);
+	} catch (e) {
+		if (e instanceof NotFoundError) {
+			return new Response('Not found', { status: 404 });
+		} else if (e instanceof MethodNotAllowedError) {
+			return new Response('Method not allowed', { status: 405 });
+		} else {
+			return new Response('Internal Error', { status: 500 });
```

Comment:
it feels like this might make it hard for people to figure out what the underlying error is

Reply:
I cant see how to figure this out. Should it throw the error that it catches on the try catch clause🤔 ?